### PR TITLE
Set explicit DB pool sizes to be in sum<100.

### PR DESCRIPTION
**Pull request by pilhuhn**

### CODE REVIEW

**Comment by theute** [comment]
FYI It fails to start:
```
﻿/usr/local/s2i/run: line 15: /opt/jboss/container/maven/default//scl-enable-maven: No such file or directory
Starting the Java application using /opt/jboss/container/java/run/run-java.sh ...
INFO exec  java -Xms63m -Xmx250m -XX:+UseParallelOldGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MaxMetaspaceSize=100m -XX:ParallelGCThreads=1 -Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -XX:CICompilerCount=2 -XX:+ExitOnOutOfMemoryError -cp "." -jar /deployments/notifications-backend-1.0.0-SNAPSHOT-runner.jar  
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2020-09-23 08:20:28,327 WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.cache.caffeine.rbac-cache.expire-after-write" was provided; it will be ignored; verify that the dependency extension for this configuration is set or you did not make a typo
2020-09-23 08:20:32,947 INFO  [org.fly.cor.int.lic.VersionPrinter] (main) Flyway Community Edition 6.5.3 by Redgate
2020-09-23 08:20:33,345 WARN  [io.agr.pool] (Agroal_1938648121) Datasource '<default>': FATAL: remaining connection slots are reserved for non-replication superuser connections
2020-09-23 08:20:33,347 ERROR [io.qua.application] (main) Failed to start application (with profile prod): org.flywaydb.core.internal.exception.FlywaySqlException: 
Unable to obtain connection from database: FATAL: remaining connection slots are reserved for non-replication superuser connections
-----------------------------------------------------------------------------------------------------------------------------------
SQL State  : 53300
Error Code : 0
Message    : FATAL: remaining connection slots are reserved for non-replication superuser connections

	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:65)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:80)
	at org.flywaydb.core.Flyway.execute(Flyway.java:453)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:158)
	at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:54)
	at io.quarkus.deployment.steps.FlywayProcessor$createBeansAndStartActions-1520831253.deploy_0(FlywayProcessor$createBeansAndStartActions-1520831253.zig:80)
	at io.quarkus.deployment.steps.FlywayProcessor$createBeansAndStartActions-1520831253.deploy(FlywayProcessor$createBeansAndStartActions-1520831253.zig:40)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:631)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:92)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
Caused by: org.postgresql.util.PSQLException: FATAL: remaining connection slots are reserved for non-replication superuser connections
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2532)
	at org.postgresql.core.v3.QueryExecutorImpl.readStartupMessages(QueryExecutorImpl.java:2644)
	at org.postgresql.core.v3.QueryExecutorImpl.<init>(QueryExecutorImpl.java:137)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:255)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:217)
	at org.postgresql.Driver.makeConnection(Driver.java:458)
	at org.postgresql.Driver.connect(Driver.java:260)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:200)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:419)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:401)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:65)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

2020-09-23 08:20:33,354 INFO  [io.sma.rea.mes.provider] (main) SRMSG00207: Cancel subscriptions
```

**Comment by pilhuhn** [comment]
I think this happens because the existing pod still holds connections open.
In this case we need to probably limit the total number to <50 so that 2 pods can talk to the DB.
And in the future even less when we have more replicas. Or change the strategy that Kube is using to replace pods (I think with 2 replicas it can be told to 1st kill one old one and replace it and then kill the 2nd old one)

**Comment by theute** [comment]
Indeed scaling down -> scaling up, works

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -14,6 +14,8 @@ quarkus.datasource.username=hook
 quarkus.datasource.password=9FLK6cMm5px8vZ52
 quarkus.datasource.jdbc.url=jdbc:postgresql://192.168.1.139:5432/notifications
 quarkus.datasource.reactive.url=vertx-reactive:postgresql://192.168.1.139:5432/notifications
+quarkus.datasource.jdbc.max-size=4
+quarkus.datasource.reactive.max-size=95
```
Comment (burmanm) [inline src/main/resources/application.properties:18]: This parameter has no effect (the quarkus.datasource.reactive.* is used only by the Quarkus' forks of drivers)

**Pull request merged**